lib/digest-auth.ts

```diff
@@ -86,11 +86,11 @@ export class DigestAuth {
 
     const parts: string[] = [];
     for (const k in authValues) {
-      if (authValues[k]) {
+      if ((authValues as any)[k]) {
```

Comment:
`as any` does not look nice. Maybe we can use `let authValue = authValues[k as keyof typeof authValues];` and then use the now typed `authValue` in the next steps.

Reply:
Good suggestion, but I will follow this guidance https://imfaber.me/typescript-how-to-iterate-over-object-properties/#use-objectentries